### PR TITLE
Update ci scripts

### PR DESCRIPTION
**Pull request by volodymyrZotov**
This PR updates the piplines to use minimum ansible-core version `2.13` as older versions have EOL(end of life) status. See [support-matrix](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix).

Versions of `ansible-core` and `ansible` in USAGEGUIDE.md and CONTRIBUTE.md was changed according to the [ansible community changelog](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-community-changelogs). But collection is still working on the `ansible-core` versions `<2.13`.

Tested the collection locally using `ansible-core 2.13.x, 2.14.x, 2.15.x` by running simple playbook `ansible-playbook 1pass_playbook.yml` and verified that it works with Connect server.

### CODE REVIEW

**Review by jillianwilson — APPROVED** [review]

File: plugins/modules/field_info.py [hunk 1]
```diff
@@ -1,7 +1,20 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
-# (c) 2021, 1Password & Agilebits (@1Password)
+#  Copyright (C) 2021  1Password & Agilebits (@1Password)
+#
+#  This program is free software: you can redistribute it and/or modify
```
Comment (verkaufer) [inline plugins/modules/field_info.py:6]: this license is incorrect, we use MIT not GNU GPL

File: scripts/run-tests.sh [hunk 2]
```diff
@@ -60,7 +60,7 @@ function inject_env_vars() {
   find ./tests/integration/ -type f -name "*.yml" -exec sed -i '' "s|__OP_CONNECT_HOST__|${OP_CONNECT_HOST}|g" {} +
   find ./tests/integration/ -type f -name "*.yml" -exec sed  -i '' "s|__OP_CONNECT_TOKEN__|${OP_CONNECT_TOKEN}|g" {} +
 
-  if [ ! -z "${OP_VAULT_ID+x}" ]; then
+  if [ -n "${OP_VAULT_ID+x}" ]; then
```
Comment (verkaufer) [inline scripts/run-tests.sh:63]: Why the switch to `-n`?

File: tests/unit/plugins/module_utils/test_item_assembly.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def test_item_creation_minimum_values():
     assert not item["urls"]
     assert not item["tags"]
     assert not item["fields"]
-    assert "sections" not in "item"
+    assert "sections" not in item
```
Comment (verkaufer) [inline tests/unit/plugins/module_utils/test_item_assembly.py:119]: nice catch!!

File: CONTRIBUTE.md [hunk 4]
```diff
@@ -14,12 +14,12 @@ After reading this document, you will:
 
 > If you have already installed a supported version of Ansible, you can skip to the [Clone the Repo](#clone-the-repo) step.
 
-This collection requires **Python v3.6 or greater**. If you don't have Python installed, consider using [pyenv](https://github.com/pyenv/pyenv) to install the supported Python runtime.
+This collection requires **Python v3.8 or greater**. If you don't have Python installed, consider using [pyenv](https://github.com/pyenv/pyenv) to install the supported Python runtime.
```
Comment (verkaufer) [inline CONTRIBUTE.md:17]: There's an inconsistency here between this file and `scripts/run-tests.sh` where we set the min python version to 3.9.  I think we want to fix the `run-tests.sh` file.

We also set `MIN_PYTHON_VERSION: "3.9"` in the GitHub action config 👀

File: plugins/modules/field_info.py (hunk 1 above)
Comment (volodymyrZotov) [reply plugins/modules/field_info.py:6]: This is a requirement from Ansible to do so. See [Collection licensing requirements](https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#id31)

If setting MIT, the sanity check will fail with the following error:
```
ERROR: Found 1 validate-modules issue(s) which need to be resolved:
ERROR: plugins/modules/item_info.py:0:0: missing-gplv3-license: GPLv3 license header not found in the first 20 lines of the module
```

Not sure if we have the ability to skip it.

**Review by verkaufer — CHANGES_REQUESTED** [review]
Please fix the license so it's MIT and not GNU GPL. 

I've also got an open question about an inconsistency in our `CONTRIBUTE.md` regarding the min Python version.

Otherwise these changes look good to me.

File: scripts/run-tests.sh (hunk 2 above)
Comment (volodymyrZotov) [reply scripts/run-tests.sh:63]: It was a sanity check error. More [details here](https://github.com/koalaman/shellcheck/wiki/SC2236).

File: CONTRIBUTE.md (hunk 4 above)
Comment (volodymyrZotov) [reply CONTRIBUTE.md:17]: According to [ansible-core support matrix](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix), the min python version will be `3.8`.

I'll update eveyrhing to mention only `3.8`

File: plugins/modules/field_info.py (hunk 1 above)
Comment (verkaufer) [reply plugins/modules/field_info.py:6]: Aha, I see :) In that case, can you please update `LICENSE.md` as well?

File: scripts/run-tests.sh (hunk 2 above)
Comment (verkaufer) [reply scripts/run-tests.sh]: understood! thanks for clarifying

File: CONTRIBUTE.md (hunk 4 above)
Comment (volodymyrZotov) [reply CONTRIBUTE.md:17]: I updated it in [this commit](https://github.com/1Password/ansible-onepasswordconnect-collection/pull/83/commits/3c9b99f0879b8863ab9f9cb8eb3a2348589468e5). But let me explain what's going on there

In order to make the tests run all ansible versions in the matrix must be compatible with the python version. Therefore the variable is not MIN_PYTHON_VERSION, but just PYTHON_VERSION which is compatible to run all the tests in the matrix

Comment (volodymyrZotov) [reply CONTRIBUTE.md:17]: In other words, the minimum python version to run the collection is 3.8, but the version to run sanity tests is 3.9

File: plugins/modules/field_info.py (hunk 1 above)
Comment (volodymyrZotov) [reply plugins/modules/field_info.py:6]: updated

File: CONTRIBUTE.md (hunk 4 above)
Comment (verkaufer) [reply CONTRIBUTE.md:17]: :+1: makes sense. Thank you for clarifying

**Review by verkaufer — APPROVED** [review]

**Comment by verkaufer** [comment]
Thanks for the changes. Looks good to me.

**Pull request merged**